codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttUnsubAckPayload.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.mqtt;
+
+import io.netty.util.internal.ObjectUtil;
+import io.netty.util.internal.StringUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Payload for MQTT unsuback message as in V5.
+ */
+public final class MqttUnsubAckPayload {
+
+    private final List<Short> unsubscribeReasonCodes;
+
+    public MqttUnsubAckPayload(short... unsubscribeReasonCodes) {
+        ObjectUtil.checkNotNull(unsubscribeReasonCodes, "unsubscribeReasonCodes");
+
+        List<Short> list = new ArrayList<Short>(unsubscribeReasonCodes.length);
+        for (Short v: unsubscribeReasonCodes) {
+            list.add(v);
+        }
+        this.unsubscribeReasonCodes = Collections.unmodifiableList(list);
+    }
+
+    public MqttUnsubAckPayload(Iterable<Short> unsubscribeReasonCodes) {
+        ObjectUtil.checkNotNull(unsubscribeReasonCodes, "unsubscribeReasonCodes");
+
+        List<Short> list = new ArrayList<Short>();
+        for (Short v: unsubscribeReasonCodes) {
+            if (v == null) {
+                break;
```

Comment:
Break or continue?

Reply:
imo makes sense to filter out nulls (continue), not stop on the first null value.

Reply:
Going to change it to throw exceptions - nulls aren't actually valid here. And if you've sent `UNSUBSCRIBE` for `N` topics, you'd expect `N` reason codes in `UNSUBACK` in MQTT 5 - having some other number is a protocol violation.